source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,37 @@
 class UrlsController < ApplicationController
+
+  # GET /urls
+  def index
+    @urls = Url.all
+    @url = Url.new
+  end
+
+  # POST /urls
+  def create
+    @url = Url.new(url_params)
+
+    respond_to do |format|
+      if @url.save
+        format.html { redirect_to root_path, notice: 'Url was successfully created.' }
+      else
+        format.html { redirect_to root_path, notice: @url.errors.full_messages }
+      end
+    end
+  end
+
+  # GET /:shortened_url
+  def redirect
+    @url = Url.where(shortened_url: params[:shortened_url]).take
```

Comment:
`take` with no argument seems very out of place to me. I think most Rails devs would expect it's alias, `first`.

[`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by) simplifies this a bit more though.